features/magic_attach.feature

```diff
@@ -0,0 +1,61 @@
+Feature: Magic attach flow related tests
+
+    @series.lts
+    @uses.config.machine_type.lxd.container
+    Scenario Outline: Attach using the magic attach flow
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I change contract to staging with sudo
+        And I create the file `/tmp/response-overlay.json` with the following:
+        """
+        {
+            "https://contracts.staging.canonical.com/v1/magic-attach": [
+            {
+              "code": 200,
+              "response": {
+                "userCode": "123",
+                "token": "testToken",
+                "expires": "expire-date",
+                "expiresIn": 2000
+              }
+            },
+            {
+                "code": 200,
+                "response": {
+                    "userCode": "123",
+                    "token": "testToken",
+                    "expires": "expire-date",
+                    "expiresIn": 2000,
+                    "contractID": "test-contract-id",
+                    "contractToken": "$behave_var{contract_token_staging}"
+                }
+            }]
+        }
+        """
+        And I append the following on uaclient config:
+        """
+        features:
+          serviceclient_url_responses: "/tmp/response-overlay.json"
+        """
+        And I run `pro attach` with sudo
+        Then stdout matches regexp:
+        """
+        Initiating attach operation...
+
+        Please sign in to advantage portal:
+        https://ubuntu.com/pro/attach
+        And provide the following user code: 123
+
+        Attaching the machine...
```

Comment:
@lechsandecki this will be the output of running `pro attach` without a token. Let us know if you think we can improve it

Reply:
I just tested the flow on staging and it works great! After my experience though there was a significant delay between seeing the "Successfully Attached Machine" message in my browser and the cli actually attaching.

A delay is inevitable, but maybe we should just add an extra line to this output, something like:

```
        Initiating attach operation...

        Please sign in to advantage portal:
        https://ubuntu.com/pro/attach
        And provide the following user code: 123

        Waiting for you to enter the code on your account... There may be a delay of several seconds
```

And maybe the 3 lines requesting action ("Please sign in ..... the following user code: 123") should be in bold?

Just ideas to consider. None of this should block the feature IMO